website/blog/2020-07-17-unnest.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Even Faster Unnest
+author: Ying Su
+authorURL: https://www.linkedin.com/in/ying-su-b00b81107/
+authorFBID: 656599427
+---
+
+Ying Su, Masha Basmanova, Orri Erling
+
+Unnest is a common operation in Facebook’s daily Presto workload. It converts an array, map or row into a flat relation. Its original implementation uses deep copy all the time and was very inefficient. In https://prestosql.io/blog/2019/08/23/unnest-operator-performance-enhancements.html, the author improved the Unnest operator by up to 10x in CPU and elapsed times by using `DictionaryBlock` when possible. In this post, we will call this implementation as "baseline". In H1 2020 we went one step further and improved it for another 5-10x. The JMH Benchmark results for different cases are shown below (The "before" means the baseline, and "after" means our new implementation):
+
+![Remote Exchange](/img/blog/2020-07-17-unnest.md/UnnestOneColumnWithoutOrdinality.png)
+![Remote Exchange](/img/blog/2020-07-17-unnest.md/UnnestOneColumnWithOrdinality.png)
+![Remote Exchange](/img/blog/2020-07-17-unnest.md/UnnestTwoColumns.png)
+![Remote Exchange](/img/blog/2020-07-17-unnest.md/UnnestRowType.png)
```

Comment:
> These images are hard to read and can't be enlarged by clicking. It would also be good to link to the JMH benchmark so readers can understand what was measured.

@aweisberg did you mean the original benchmark results in its original format? Or a link to the benchmark source code?
I enlarged the fonts in these images. These images looked vague because I took screen shot on the whole page. The real blog should display them better. Also added `The benchmark can be found in presto-main/src/test/java/com/facebook/presto/operator/BenchmarkUnnestOperator.java.`

Reply:
Hey the images look a bit unclear on the actual page. I am running the web site locally with yarn start. I was thinking it would be nice if the image could be a link to a larger sharper version.

I was thinking link to the actual benchmark.